Filter reflected classes by class type instead of searching for '=> ' tags

Grifs opened the pull request:
In Scala 2.12, the '=> ' tag was a reliable way to filter, but with Scala 2.13 this is no longer present. Filtering for "NullaryMethodType" instead gives the desired functionality now.

Improve the syntax of the mapping of a tuple and unapply to 2 names.



No changelog entry added as I consider it part of updating to Scala 2.13

rcannood reviewed (changes requested):
Could you add an entry to the changelog

Grifs commented:
> Could you add an entry to the changelog

I considered this change a part of "`Scala`: Updated to Scala 2.13 and updated several dependencies." (or it should have been) from this release

Grifs commented on src/main/scala/io/viash/schemas/CollectedSchemas.scala:
```diff
@@ -80,13 +80,13 @@ object CollectedSchemas {
-          .filter(m => !m.info.toString.startsWith("=> ") || x._2 != 0) // Only regular members if base class, otherwise all members
-          .map(y => MemberInfo(y, (constructorMembers.contains(y.shortName)), x._1.fullName, x._2))
-      )
+           .filter(m => !m.info.getClass.toString.endsWith("NullaryMethodType") || index != 0) // Only regular members if base class, otherwise all members
```
Good catch!

rcannood commented:
> > Could you add an entry to the changelog
> 
> I considered this change a part of "`Scala`: Updated to Scala 2.13 and updated several dependencies." (or it should have been) from this release

Fair enough!
